scripts/release.js

```diff
@@ -0,0 +1,43 @@
+const path = require("path");
+const { exec, getExecOutput } = require("@actions/exec");
+
+const { version } = require("../package.json");
+const tag = `v${version}`;
+const releaseLine = `v${version.split(".")[0]}`;
+
+process.chdir(path.join(__dirname, ".."));
+
+(async () => {
+  const { exitCode, stderr } = await getExecOutput(
+    `git`,
+    ["ls-remote", "--exit-code", "origin", "--tags", `refs/tags/${tag}`],
+    {
+      ignoreReturnCode: true,
+    }
+  );
+  if (exitCode === 0) {
+    console.log(
+      `Action is not being published because version ${tag} is already published`
+    );
+    return;
+  }
+  if (exitCode !== 2) {
+    throw new Error(`git ls-remote exited with ${exitCode}:\n${stderr}`);
+  }
+
+  await exec("yarn", ["build"]);
+
+  await exec("git", ["checkout", "--detach"]);
+  await exec("git", ["add", "--force", "dist"]);
+  await exec("git", ["commit", "-m", tag]);
+
+  await exec("changeset", ["tag"]);
+
+  await exec("git", [
+    "push",
+    "--force",
+    "--follow-tags",
+    "origin",
+    `HEAD:refs/heads/${releaseLine}`,
+  ]);
```

Comment:
note that i've chosen to use a branch for this because branches sync better than tags between remote and a local copy, to push out a tag with the same label we need to first destroy it and recreate. In general, I prefer to treat tags as immutable pointers.